reactor-core/src/test/java21/reactor/core/scheduler/LoomBoundedElasticSchedulerTest.java

```diff
@@ -0,0 +1,267 @@
+package reactor.core.scheduler;
+
+import java.util.Queue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import reactor.core.Disposable;
+import reactor.test.util.RaceTestUtils;
+import reactor.util.concurrent.Queues;
+
+class LoomBoundedElasticSchedulerTest {
+
+	Scheduler scheduler;
+
+	@BeforeEach
+	void setup() {
+		scheduler = new LoomBoundedElasticScheduler(2,
+				3,
+				Thread.ofVirtual()
+				      .name("loom", 0)
+				      .factory());
+		scheduler.init();
+	}
+
+	@AfterEach
+	void teardown() {
+		scheduler.dispose();
+	}
+
+	@Test
+	public void ensuresTasksScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(1);
+
+		Disposable disposable = scheduler.schedule(latch::countDown);
+
+		Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		Assertions.assertThat(disposable.isDisposed()).isTrue();
+	}
+
+	@Test
+	public void ensuresTasksDelayedScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(1);
+
+		Disposable disposable = scheduler.schedule(latch::countDown, 200, TimeUnit.MILLISECONDS);
+
+		Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		Assertions.assertThat(disposable.isDisposed()).isTrue();
+	}
+
+	@Test
+	public void ensuresTasksDelayedZeroDelayScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(1);
+
+		Disposable disposable = scheduler.schedule(latch::countDown, 0, TimeUnit.MILLISECONDS);
+
+		Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		Assertions.assertThat(disposable.isDisposed()).isTrue();
+	}
+
+	@Test
+	public void ensuresTasksPeriodicScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(10);
+
+		Disposable disposable = scheduler.schedulePeriodically(latch::countDown,
+				100,
+				10,
+				TimeUnit.MILLISECONDS);
+
+		Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		Assertions.assertThat(disposable.isDisposed()).isFalse();
+		disposable.dispose();
+		Assertions.assertThat(disposable.isDisposed()).isTrue();
+	}
+
+	@Test
+	public void ensuresTasksPeriodicZeroInitialDelayScheduling() throws InterruptedException {
```

Comment:
This and the similar tests don't really validate whether the time is applied correctly. Can we do something about validating the arguments have any actual impact? The assertions stay the same, they should be different if the arguments differ.